features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Shared/AppsAndKeys/SubscribeToApi.jsx

```diff
@@ -102,8 +104,11 @@ const subscribeToApi = (props) => {
 
     useEffect(() => {
         if (applicationsAvailable && applicationsAvailable[0]) {
-            seAapplicationsList(applicationsAvailable);
+            setAapplicationsList(applicationsAvailable);
             setAppSelected(applicationsAvailable[0].value);
+            const newRequest = { ...subscriptionRequest };
+            newRequest.applicationId = applicationsAvailable[0].value;
+            updateSubscriptionRequest(newRequest);
```

Comment:
shouldn't there be a null check before invoking the `updateSubscriptionRequest()` method? Since it's passed from props.

Reply:
I don't think it's required since we are passing props correctly, If someone is reusing the component and he fails to pass the props, It should break from there otherwise it will be very hard to detect an issue like that. Since we are also doing prop validation that error will also be printed if we are not passing the prop